esphome/components/micronova/switch/__init__.py

```diff
@@ -0,0 +1,52 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import switch
+from esphome.const import (
+    ICON_POWER,
+)
+
+from .. import (
+    MicroNova,
+    MicroNovaFunctions,
+    CONF_MICRONOVA_ID,
+    CONF_MEMORY_LOCATION,
+    CONF_MEMORY_ADDRESS,
+    MICRONOVA_LISTENER_SCHEMA,
+    micronova_ns,
+)
+
+CONF_STOVE = "stove"
+CONF_MEMORY_DATA_ON = "memory_data_on"
+CONF_MEMORY_DATA_OFF = "memory_data_off"
+
+TYPES = [
+    CONF_STOVE,
+]
+
+MicroNovaSwitch = micronova_ns.class_("MicroNovaSwitch", switch.Switch, cg.Component)
+
+CONFIG_SCHEMA = cv.Schema(
+    {
+        cv.GenerateID(CONF_MICRONOVA_ID): cv.use_id(MicroNova),
+        cv.Optional(CONF_STOVE): switch.switch_schema(
+            MicroNovaSwitch,
+            icon=ICON_POWER,
+        )
+        .extend(MICRONOVA_LISTENER_SCHEMA)
+        .extend({cv.Optional(CONF_MEMORY_DATA_OFF, default=0x06): cv.hex_int_range()})
+        .extend({cv.Optional(CONF_MEMORY_DATA_ON, default=0x01): cv.hex_int_range()}),
```

Comment:
These should be in a single extend
```suggestion
        .extend(
            {
                cv.Optional(CONF_MEMORY_DATA_OFF, default=0x06): cv.hex_int_range(),
                cv.Optional(CONF_MEMORY_DATA_ON, default=0x01): cv.hex_int_range(),
            }
        ),
```

Reply:
Done.